libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -35,9 +36,16 @@ export class BookSearchComponent implements OnInit, OnDestroy {
   }
 
   ngOnInit(): void {
-    this.store.select(getAllBooks).pipe(takeUntil(this.notifier)).subscribe(books => {
-      this.books = books;
-    });
+    this.searchForm.valueChanges
+      .pipe(takeUntil(this.notifier))
+      .pipe(debounceTime(500), distinctUntilChanged())
+      .subscribe((key) => {
```

Comment:
Just a small suggestion. `takeUntil` should mostly always be the last operator (there are exceptions). I suggest reading this article: https://cartant.medium.com/rxjs-avoiding-takeuntil-leaks-fb5182d047ef